tests/test_ir.py

```diff
@@ -1167,3 +1167,23 @@ def test_ssa_get_on_op():
         ValueError, match="SSAValue.get: expected operation with a single result."
     ):
         SSAValue.get(op2)
+
+
+def test_repr_operation():
+    """Test that repr functions on operations with uses do not create excessive recursion."""
+    op1 = test.TestOp(result_types=[i32])
+    op2 = test.TestOp(result_types=[i32])
+    op3 = test.TestOp(result_types=[i32])
+
+    op4 = test.TestOp(operands=[op1, op2, op3] * 1000, result_types=[i64])
+
+    op5 = test.TestOp(operands=[op1, op2, op3, op4])
+
+    # This is not designed to test that the repr takes a particular form.
+    # It just tests that the result is returned without generating maximum recursion depth errors
+    try:
+        result = repr(op5)
+    except RecursionError:
+        pytest.fail("repr method on Operation reaches recursion limit.")
+
+    assert bool(result)
```

Comment:
Is there a way to test things more directly? I think testing the value of the repr here would work just as well in the sense of demonstrating non-recursiveness.

Reply:
Sorry, I don't understand. I'll clone this locally and try to make sense of it. I have a feeling we might be using different meanings for recursive. Am I understanding correctly that you mean that you can have the same method without recursive function calls? I mean that the repr of operation doesn't contain the repr of another operation.

Reply:
I think I fixed it locally for me in a way that should make both of us happy. I just opened #5434, after that's merged I'll open a comprehensive repr refactor PR.